In [12]:
f = open("archivo.txt","r")

contenido = f.read() 
print(contenido)
f.close()

1 2
2 4
3 6
4 8


In [13]:
f = open("archivo.txt","r")

contenido = f.read(9) #imprime 9 caracteres, los saltos de linea los cuenta como caracter 
print(contenido)
f.close()

1 2
2 4
3


In [3]:
f = open("archivo.txt","r")
print(f.readline()) #imprime solo una linea cada vez que lo invocas
print(f.readline())

f.close()

# for i in range(0,4):
#     print(f.readline())

1 2

2 4



In [1]:
# archivo = "archivo.txt"
# text = open(archivo).readlines()

f = open("archivo.txt")
text = f.readlines() #el metodo .readlines() mete el texto en un array, y cada linea en un indice

print("tipo ",type(text)) #me doy cuenta de que .readlines() lo castea a list

numLineas = len(text) #con esto puedo contar cuantas lineas tiene, xq cuenta la longitud del array
print("numero de lineas", numLineas)

print("--1--")
print(text)

print("--2--")
for line in text:
    print(line.split()) #me doy cuenta de que split() divide cada indice de la lista y lo convierte en una lista nueva
                     #por defecto usa como separador el espacio, pero le puedo pasar como argumento el separador q quiera
                     #tambien me doy cuenta de que en vez de "line" puedo poner simplemente una "i"

print("--3--")
for line in text:
    print(line)
    
print("--4--")
for line in text:
    print(line,end="")
    
f.close()

tipo  <class 'list'>
numero de lineas 4
--1--
['1 2\n', '2 4\n', '3 6\n', '4 8']
--2--
['1', '2']
['2', '4']
['3', '6']
['4', '8']
--3--
1 2

2 4

3 6

4 8
--4--
1 2
2 4
3 6
4 8

In [4]:
from math import pi

print("%0.4f" % pi)
print("%0.5f" % pi)
print("%0.6f" % pi)

3.1416
3.14159
3.141593


In [11]:
#ESTE SIRVE PARA ACOTAR STRING, %s (of string)
print ("%.4s" % "hola mundo") 

#ESTE SIRVE PARA ACOTAR NUMEROS, %f (of float)
from math import pi
print("%.4f" % pi)

hola
3.1416


<hr>
<b><font color='red'>Ejercicio 01</font></b>  

Lee el archivo.txt 

In [15]:
f = open("archivo.txt", "r")

contenido = f.read()

print(contenido)

f.close()


1 2
2 4
3 6
4 8


y haz que tenga la siguiente salida:

    1.0   2.00  
    2.0   4.00  
    3.0   6.00  
    4.0   8.00  


In [31]:
f = open("archivo.txt", "r")

contenido = f.readlines()

for i in contenido:
#     num = ("%0.2f"%i)
    print("  ",i,end="")
# print(contenido)

f.close()

   1 2
   2 4
   3 6
   4 8

In [53]:
f = open("archivo.txt", "r")

contenido = f.read()

lista=[]
for i in contenido: #hago una lectura de todo el fichero, si es un digit lo guardo en un array
    if i.isdigit():
        lista.append(i)
print(lista)

contador=1
for i in lista: #aqui lo que hago es tratar a la lista como me de la gana
    if contador%2:
        print("")
    i = int(i)
    print("  ","%0.2f"%i, end="  ")
    contador+=1

f.close()

['1', '2', '2', '4', '3', '6', '4', '8']

   1.00     2.00  
   2.00     4.00  
   3.00     6.00  
   4.00     8.00  

<hr>
<b><font color='red'>Ejercicio 02</font></b>  

Haz lo mismo con el archvo archivo_error.txt,con ciddado de que a este archivo le falta algún dato, haz que en ese caso cuando lo lea, si no hay dos columnas no lo escriba.

In [96]:
f = open("archivo_error.txt","r")
contenido = f.readlines()

lista=[]
for line in contenido:
    if len(line.split()) == 2:
#         print(line.split())
        lista.append(line.split()[0])
        lista.append(line.split()[1])
        
# print(lista)
conatdor=1
for i in lista:
    if contador%2:
        print("")
    i = int(i)
    print(" ","%0.2f"%i,end=" ")
    contador+=1


  1.00   2.00 
  3.00   6.00 
  4.00   8.00 

<hr>
<b><font color='red'>Ejercicio 03</font></b>  

Lee el archivo.txt, haz que imprima las columnas en las filas y las filas en las columnas

In [120]:
f = open("archivo.txt")
contenido = f.readlines()
    
print(contenido)
for i in contenido:
    print(i.split()[0],end=" ")
print("")
for i in contenido:
    print(i.split()[1],end=" ")

f.close()

1 2

2 4

3 6

4 8
['1 2\n', '2 4\n', '3 6\n', '4 8']
['1', '2']
['2', '4']
['3', '6']
['4', '8']
['1 2\n', '2 4\n', '3 6\n', '4 8']
1 2 3 4 
2 4 6 8 

<hr>
<b><font color='red'>Ejercicio 04</font></b>  

1. Escribe en un archivo la siguiente matriz:
$
\begin{equation}
\begin{pmatrix}
3 & 1 & 3\\
-1 & 1 & 0\\
-2 & 4 & 1
\end{pmatrix}
\end{equation} 
$
2. Lee el archivo
3. Calcula con los datos leidos el determinante de la matriz y lo saque por pantalla

In [139]:
f = open("matriz.txt")
contenido = f.readlines()

print(contenido)

x=[]
for line in contenido:
#     lista.append(line)
    print(line.split())

# print(lista)
f.close()

['3 1 3\n', '-1 1 0\n', '-2 4 1']
['3', '1', '3']
['-1', '1', '0']
['-2', '4', '1']


<hr>
<b><font color='red'>Ejercicio 05</font></b>  

En el criptoanálisis, la técnica de análisis de frecuencias consiste en el aprovechamiento de estudios sobre la frecuencia de las letras o grupos de letras en los idiomas para poder establecer hipótesis para aprovecharlas para poder descifrar un texto cifrado sin tener la clave de descifrado (romper). Es un método típico para romper cifrados clásicos. 
Haz una anailisis de frecuencias para los 10 primeros càpitulos de el libro "el_quijote.txt"

In [200]:
abc="ABCDEFGHIJKLMNÑOPQRSTUVWXYZ"

fichero = open("el_quijote.txt",encoding="utf8")
contenido = fichero.read()

s = []
cont = 0
for i in contenido:
    s.append(i.upper().replace("Á","A").replace("É","E").replace("Í","I").replace("Ó","O").replace("Ú","U"))
    if i.isalpha(): #esto era lo que me faltaba comprobar, comprobar que sea una letra
        cont+=1
    
print("contador de solamente las letras: ")
print(cont)

#en este array guardo los numeros decima
f=[]
for i in abc:
    f.append(0)
    
for i in s:
    for j in range(27):
        if i==abc[j]:
            f[j]+=1

print("veces que se repite cada letra:")
print(f)

f2=[]
for i in f:
    f2.append(i/cont)
#     print(i/cont)

for i in abc:
    print("%.2c  " % i,"%.3f" % f2[(abc.index(i))])

fichero.close()

contador de solamente las letras: 
94079
veces que se repite cada letra:
[11975, 1609, 3434, 5001, 13021, 449, 1001, 1110, 4950, 619, 1, 5627, 2480, 6222, 0, 9313, 1990, 1825, 5765, 6882, 3446, 4554, 1015, 0, 15, 1370, 405]
A   0.127
B   0.017
C   0.037
D   0.053
E   0.138
F   0.005
G   0.011
H   0.012
I   0.053
J   0.007
K   0.000
L   0.060
M   0.026
N   0.066
Ñ   0.000
O   0.099
P   0.021
Q   0.019
R   0.061
S   0.073
T   0.037
U   0.048
V   0.011
W   0.000
X   0.000
Y   0.015
Z   0.004


Salida:

    A   0.127
    B   0.017
    C   0.037
    D   0.053
    E   0.138
    F   0.005
    .
    .
    .

<hr>
<b><font color='red'>Ejercicio 06</font></b>  

haz un analisis de frecuencias en el libro encriptado "RH RQCEPESEFM AR MPM.txt", y comparala con el analisis anterior


In [204]:
abc="ABCDEFGHIJKLMNÑOPQRSTUVWXYZ"

fichero = open("RH RQCEPESEFM AR MPM.txt",encoding="utf8")
contenido = fichero.read()

s = []
cont = 0
for i in contenido:
    s.append(i.upper().replace("Á","A").replace("É","E").replace("Í","I").replace("Ó","O").replace("Ú","U"))
    if i.isalpha(): #esto era lo que me faltaba comprobar, comprobar que sea una letra
        cont+=1
    
print("contador de solamente las letras: ")
print(cont)

#en este array guardo los numeros decima
f=[]
for i in abc:
    f.append(0)
    
for i in s:
    for j in range(27):
        if i==abc[j]:
            f[j]+=1

print("veces que se repite cada letra:")
print(f)

f3=[]
for i in f:
    f3.append(i/cont)
#     print(i/cont)

# for i in abc:
#     print("%.2c  " % i,"%.3f" % f3[(abc.index(i))])
    
print("el_quijote.txt","  RH RQCEPESEFM AR MPM.txt")    
for i in abc:
    print("%.2c  " % i,"%.3f" % f2[(abc.index(i))],"       %.2c  " % i,"%.3f" % f3[(abc.index(i))])

fichero.close()

contador de solamente las letras: 
58820
veces que se repite cada letra:
[2909, 375, 2354, 3576, 7410, 410, 8, 3066, 1907, 701, 4020, 86, 5494, 1593, 666, 571, 3882, 4080, 7983, 1017, 2615, 2655, 529, 19, 120, 546, 222]
el_quijote.txt   RH RQCEPESEFM AR MPM.txt
A   0.127        A   0.049
B   0.017        B   0.006
C   0.037        C   0.040
D   0.053        D   0.061
E   0.138        E   0.126
F   0.005        F   0.007
G   0.011        G   0.000
H   0.012        H   0.052
I   0.053        I   0.032
J   0.007        J   0.012
K   0.000        K   0.068
L   0.060        L   0.001
M   0.026        M   0.093
N   0.066        N   0.027
Ñ   0.000        Ñ   0.011
O   0.099        O   0.010
P   0.021        P   0.066
Q   0.019        Q   0.069
R   0.061        R   0.136
S   0.073        S   0.017
T   0.037        T   0.044
U   0.048        U   0.045
V   0.011        V   0.009
W   0.000        W   0.000
X   0.000        X   0.002
Y   0.015        Y   0.009
Z   0.004        Z   0.004


salida:

    el_quijote.txt   RH RQCEPESEFM AR MPM.txt
    A   0.127        A   0.049
    B   0.017        B   0.006
    C   0.037        C   0.040
    .                .
    .                .
    .                .